src/plugin/manager.cc

```diff
@@ -241,6 +241,30 @@ namespace Astroid {
     return clrs;
   }
 
+  std::vector<std::pair<ustring, ustring>> PluginManager::AstroidExtension::get_queries () {
+    std::vector<std::pair<ustring, ustring>> queries = {};
+
+    if (!active || astroid->plugin_manager->disabled) return queries;
+
+    for (PeasPluginInfo * p : astroid->plugin_manager->astroid_plugins) {
+      PeasExtension * pe = peas_extension_set_get_extension (extensions, p);
+
+      GList * list = astroid_activatable_get_queries (ASTROID_ACTIVATABLE(pe));
+
+      if (list != NULL) {
+        std::vector<ustring> _list = Glib::ListHandler<ustring>::list_to_vector (list, Glib::OWNERSHIP_NONE);
```

Comment:
What happens with the ownership of `list` is it cleaned up?

Reply:
I have no idea. I think I'm doing the exact same thing as what happens in `PluginManager:AstroidExtension::get_tag_colors`, which I assumed did everything right memory-management/ownership-wise.

I just ran astroid in Valgrind and it seems my assumption was wrong, get_queries() does leak memory. This probably means that other `PluginManager::AstroidExtension::get_*` functions also leak memory (but I could be wrong, I'm pretty bad at C++).

Reply:
I added a `g_list_free(list)` call, please let me know if you think this is enough.

Reply:
Looks good, you are probably right that there is a bug i the other methods.